src/fides/api/service/dataset/dataset_service.py

```diff
@@ -0,0 +1,155 @@
+
+    return True, None
+
+
+def get_identities_and_references(
+    dataset_config: DatasetConfig,
+) -> Set[str]:
+    """
+    Returns all identity and dataset references in the dataset.
+    """
+    result: Set[str] = set()
+    dataset: GraphDataset = dataset_config.get_graph()
+    for collection in dataset.collections:
+        # Process the identities in the collection
+        result.update(collection.identities().values())
+        for _, field_refs in collection.references().items():
+            # Take first reference only
+            ref, edge_direction = field_refs[0]
+            if edge_direction == "from" and ref.dataset != dataset_config.fides_key:
+                result.add(ref.value)
+    return result
+
+
+def run_test_access_request(
+    db: Session,
+    policy: Policy,
+    dataset_config: DatasetConfig,
+    input_data: Dict[str, Any],
+) -> PrivacyRequest:
+    """
+    Creates a privacy request with a source of "Dataset test" that runs an access request for a single dataset.
+    The input data is used to mock any external dataset values referenced by our dataset so that it can run
+    complete isolation.
+    """
+
+    # Create a privacy request with a source of "Dataset test"
+    # so it's not shown to the user.
+    privacy_request = PrivacyRequest.create(
+        db=db,
+        data={
+            "policy_id": policy.id,
+            "source": PrivacyRequestSource.dataset_test,
+            "status": PrivacyRequestStatus.in_processing,
+        },
+    )
+
+    # Remove periods and colons to avoid them being parsed as path delimiters downstream.
+    escaped_input_data = {
+        key.replace(".", "_").replace(":", "_"): value
+        for key, value in input_data.items()
+    }
+
+    # Manually cache the input data as identity data.
+    # We're doing a bit of trickery here to avoid asking for labels for custom identities.
+    predefined_fields = Identity.model_fields.keys()
+    input_identity = {
+        key: (
+            value
+            if key in predefined_fields
+            else LabeledIdentity(label=key, value=value)
+        )
+        for key, value in escaped_input_data.items()
+    }
+    privacy_request.cache_identity(input_identity)
+
+    # We're providing external reference values directly, so we can replace
+    # field references with identities and provide the values as seed data
+    # at the start of the traversal. This way we don't need to run collections
+    # from any other datasets, only from the dataset we're testing.
+    graph_dataset = dataset_config.get_graph()
+    for collection in graph_dataset.collections:
+        for field in collection.fields:
+            for ref, edge_direction in field.references[:]:
+                if edge_direction == "from" and ref.dataset != dataset_config.fides_key:
+                    field.identity = f"{ref.dataset}_{ref.collection}_{'_'.join(ref.field_path.levels)}"
+                    field.references.remove((ref, "from"))
```

Comment:
can we extract this into a separate function? Since we're mutating the `graph_dataset` var, we can just have the function return that